tests/run.sh

```diff
@@ -17,10 +17,14 @@ if [ "$CI" = true ]; then
     set -x
 fi
 
-source_dir="$PWD"
+source_dir="$(dirname "$(realpath "$(type -p $0)")")"/..
 build_dir="$CMAKE_BINARY_DIR"
 if [ -z "$build_dir" ]; then
-    build_dir="$PWD/build"
+    if [ -d "$source_dir/build" ]; then
+        build_dir="$source_dir/build"
+    else
+        build_dir="$PWD"
+    fi
```

Comment:
Did this commit end up in this PR on purpose? My comments from the last time that I saw this patch still apply (e.g. bashism that doesn't work in zsh; I don't see how this worked before my patch, ...)